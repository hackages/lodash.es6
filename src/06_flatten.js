/**
 * Flattens array a single level deep.
 */
export function flatten(array) {}
