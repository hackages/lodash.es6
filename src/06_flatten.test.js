import {flatten} from './06_flatten'

describe('flatten', () => {
  it('should flatten', () => {
    expect(
      flatten([1, 2, false, {a: 32}, NaN, null, undefined])
    ).toMatchSnapshot()

    expect(
      flatten([[1, 2], [[false, {a: 32}], NaN], [null, [{undefined}]]])
    ).toMatchSnapshot()

    expect(flatten([[[[[[[[[[[[['Sorry']]]]]]]]]]]]])).toMatchSnapshot()
  })
})
