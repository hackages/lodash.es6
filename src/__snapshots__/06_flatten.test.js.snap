// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flatten should flatten 1`] = `
Array [
  1,
  2,
  false,
  Object {
    "a": 32,
  },
  NaN,
  null,
  undefined,
]
`;

exports[`flatten should flatten 2`] = `
Array [
  1,
  2,
  Array [
    false,
    Object {
      "a": 32,
    },
  ],
  NaN,
  null,
  Array [
    Object {
      "undefined": undefined,
    },
  ],
]
`;

exports[`flatten should flatten 3`] = `
Array [
  Array [
    Array [
      Array [
        Array [
          Array [
            Array [
              Array [
                Array [
                  Array [
                    Array [
                      Array [
                        "Sorry",
                      ],
                    ],
                  ],
                ],
              ],
            ],
          ],
        ],
      ],
    ],
  ],
]
`;
